packages/typespec-ts/test/modularUnit/modelsGenerator.spec.ts

```diff
@@ -1126,6 +1135,403 @@ describe("inheritance & polymorphism", () => {
+      modelFile?.getFullText()!,
+      `
+        export interface Bar {
+          prop1: string;
+          prop2: number;
+        }`
+    );
+    const optionFile = await emitModularModelsFromTypeSpec(tspContent, true);
+    assert.ok(optionFile);
+    assertEqualContent(
+      optionFile?.getFullText()!,
+      `
+      import { OperationOptions } from "@azure-rest/core-client";
+      
+      export interface ReadOptions extends OperationOptions {
+        prop3?: Date;
+        prop5?: Bar;
+      }`
+    );
+    const operationFiles = await emitModularOperationsFromTypeSpec(tspContent);
+    assert.ok(operationFiles);
+    assert.equal(operationFiles?.length, 1);
+    assertEqualContent(
+      operationFiles?.[0]?.getFullText()!,
+      `
+      import { TestingContext as Client } from "../rest/index.js";
+      import {
+        StreamableMethod,
+        operationOptionsToRequestParameters,
+      } from "@azure-rest/core-client";
+      export function _readSend(
+        context: Client,
+        pathParam: string,
+        queryParam: string,
+        prop1: string,
+        prop2: number,
+        prop4: string,
+        options: ReadOptions = { requestOptions: {} }
+      ): StreamableMethod<Read200Response> {
+        return context
+          .path("/{pathParam}", pathParam)
+          .post({
+            ...operationOptionsToRequestParameters(options),
+            queryParameters: { queryParam: queryParam },
+            body: {
+              prop1: prop1,
+              prop2: prop2,
+              prop3: options?.prop3?.toISOString(),
+              prop4: prop4,
+              prop5: {
+                prop1: options?.prop5?.["prop1"],
+                prop2: options?.prop5?.["prop2"],
+              },
+            },
+          });
+      }
+      export async function _readDeserialize(result: Read200Response): Promise<void> {
+        if (result.status !== "200") {
+          throw result.body;
+        }
+        return;
+      }
+      export async function read(
+        context: Client,
+        pathParam: string,
+        queryParam: string,
+        prop1: string,
+        prop2: number,
+        prop4: string,
+        options: ReadOptions = { requestOptions: {} }
+      ): Promise<void> {
+        const result = await _readSend(
+          context,
+          pathParam,
+          queryParam,
+          prop1,
+          prop2,
+          prop4,
+          options
+        );
+        return _readDeserialize(result);
+      }`,
+      true
+    );
+  });
+
+  it("should flatten alias if spread in the payload with optional parameters", async () => {
+    const tspContent = `
+    alias Foo = {
+      @path
+      prop1: string;
+      prop2: int64;
+      prop3?: utcDateTime;
+      @query
+      prop4: offsetDateTime;
+      prop5?: Bar;
+    };
+    model Bar {
+      prop1: string;
+      prop2: int64;
+    }
+    op read(@path pathParam: string, ...Foo, @query queryParam: string): OkResponse;
```

Comment:
Here, I put the alias in the middle, and path and query related parameter can be order by typespec correctly, but as there's no way to know the order between bodyParameter and other parameters, I generate the body flattened parameter prop2 to the last.